lib/matplotlib/axis.py

```diff
@@ -154,6 +140,11 @@ def __init__(
             grid_alpha = mpl.rcParams["grid.alpha"]
         grid_kw = {k[5:]: v for k, v in kwargs.items()}
 
+        remaining_kwargs = {
+            k: v for k, v in kwargs.items() if not k.startswith('grid_')}
+        if remaining_kwargs:
+            raise ValueError(f"Invalid kwargs: {remaining_kwargs.keys()}")
```

Comment:
This probably should start as a warning for a cycle?  On one hand I hope these is no usage of this in the wild (it is not documented and users should not be making `Tick` directly!), but on the other we tend to be very conservative about this sort of thing.

Eitherway, this needs an API change note.

Reply:
Where and what should I write about in the API change note?
Should I write about changing it to raise in the next cycle?